File: 20190917/20190925/747. 至少是其他数字两倍的最大数.cc
/*
 * Description: 至少是其他数字两倍的最大数
 * Author: AskeyNil
 * CreateDate: 2019-09-25 07:45:22
 * LastEditors: AskeyNil
 *
 * *********************************
 * **                             **
 * **     　  你只有足够努力     　  **
 * **       才能看上去毫不费力       **
 * **                             **
 * *********************************
 *
 */

#include <iostream>
#include <vector>

using namespace std;

class Solution {
  public:
    int dominantIndex(vector<int> &nums) {
        int i = -1, max_number = -2;
        for (int index = 0; index < nums.size(); index++) {
            if (max_number == -2) {
                i = index;
                max_number = nums[index];
            } else {
                if (max_number < nums[index]) {
                    if (max_number * 2 <= nums[index]) {
                        i = index;
                    } else {
                        i = -1;
                    }
                    max_number = nums[index];
                } else if (nums[index] * 2 > max_number) {
                    i = -1;
                }
            }
        }
        return i;
    }
};

File: 20190917/20190925/747. 至少是其他数字两倍的最大数.py
'''
Description: 至少是其他数字两倍的最大数
Author: AskeyNil
CreateDate: 2019-09-25 07:45:18
LastEditors: AskeyNil

*********************************
**                             **
**     　  你只有足够努力     　  **
**       才能看上去毫不费力       **
**                             **
*********************************

'''


class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        i = -1
        max_number = None
        for index, num in enumerate(nums):
            if max_number == None:
                i = index
                max_number = num
            else:
                if max_number < num:
                    max_number = num
                    if max_number * 2 < num:
                        i = index
                    else:
                        i = -1
        return i
